device/globals/global_vm.gd

```diff
@@ -146,24 +148,33 @@ func update_camera(time):
 
 	var cpos = camera.get_position()
 
+	# The camera position is set to target when it's about to overstep it,
+	# or when it's moved there instantly.
+	# Compare the camera and target position until then
 	if cpos != pos:
-		#return
+		var v = pos - cpos  # vector to move along
+		var step = cam_speed * time  # pixel size of step to move
 
-		var dif = pos - cpos
-		var dist = cam_speed * time
-		if dist > dif.length() || cam_speed == 0:
+		# This is where we may overstep or move instantly
+		if step > v.length() || cam_speed == 0:
 			camera.set_position(pos)
-			#return
 		else:
-			camera.set_position(cpos + dif.normalized() * dist)
-			pos = cpos + dif.normalized() * dist
+			pos = cpos + v.normalized() * step
+			camera.set_position(pos)
 
 	var half = game_size / 2
-	pos = _adjust_camera(pos)
+
+	clamp_data = _clamp_camera(pos, half)
+	camera_clamped = clamp_data[0]
+	pos = clamp_data[1]
+
 	var t = Transform2D()
-	t[2] = (-(pos - half))
+	t.origin = (-(pos - half))
```

Comment:
Could `(-(pos - half))` be replaced with `-(pos - half)`?

Reply:
Hold on, isn't this the same as `t.origin = half - pos`? Would that be cleaner?

This is most likely the line that will require knowledge when I make a plz halp pull request, so I'll defer judgement to you entirely.

Reply:
Yes, unless it's a one-element tuple, which I don't think GDScript has, so I think you're right. Come to think of it, I'm pretty sure it's a Vector2, so that should be correct.

Reply:
Also, wouldn't it be better to either use an array/dictionary or two scalar instead of both?

Reply:
The pleasures of weak typing ;)

`pos` is a Vector2 and so's `half` by virtue of coming from `Viewport.size`.

I don't understand "Also, wouldn't it be better to either use an array/dictionary or two scalar instead of both?" but we can take that up on IRC after I force-push `half - pos`.